packages/openassistants/openassistants/functions/crud.py

```diff
@@ -86,3 +97,68 @@ def read(self, slug: str) -> Optional[IBaseFunction]:
 
     def list_ids(self) -> List[str]:
         return [function.get_id() for function in self.functions]
+
+
+class OpenAPICRUD(PythonCRUD):
+    openapi: OpenAPISpec
```

Comment:
For this to be practically useful, you probably need to provide some sort of authorization solution as well, but good start!

Reply:
Yeah I think auth is a good extension for a v2 iteration 🙌